sysdig/resource_sysdig_secure_cloud_auth_account_test.go

```diff
@@ -107,7 +110,210 @@ resource "sysdig_secure_cloud_auth_account" "sample-1" {
+			{
+				Config: fmt.Sprintf(`
+		resource "sysdig_secure_cloud_auth_account" "gcp_agentless_scanning" {
+			provider_id   = "gcp-agentless-test-%s"
+			provider_type = "PROVIDER_GCP"
+			enabled       = true
+
+		    feature {
+              secure_config_posture {
+              enabled    = true
+                components = ["COMPONENT_SERVICE_PRINCIPAL/secure-posture"]
+              }
+
+			  secure_agentless_scanning {
+			    enabled    = true
+			    components = ["COMPONENT_SERVICE_PRINCIPAL/secure-scanning"]
+			  }
+		    }
+
+            component {
+				type     = "COMPONENT_SERVICE_PRINCIPAL"
+				instance = "secure-posture"
+				service_principal_metadata = jsonencode({
+				  gcp = {
+					key = "%s"
+				  }
+				})
+            }
+
+			component {
+				type                       = "COMPONENT_SERVICE_PRINCIPAL"
+				instance                   = "secure-scanning"
+				service_principal_metadata = jsonencode({
+					gcp = {
+						workload_identity_federation = {
+							pool_provider_id = "pool_provider_id_value"
+						}
+						email = "email_value"
+					}
+				})
+			}
+		}`, accountID, getEncodedServiceAccountKey("gcp-agentless-test", accountID)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "provider_type", "PROVIDER_GCP"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "enabled", "true"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "feature.0.secure_agentless_scanning.0.enabled", "true"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "feature.0.secure_agentless_scanning.0.components.0", "COMPONENT_SERVICE_PRINCIPAL/secure-scanning"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "component.0.type", "COMPONENT_SERVICE_PRINCIPAL"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "component.0.instance", "secure-posture"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "component.0.service_principal_metadata", "{\"gcp\":{\"key\":\""+getEncodedServiceAccountKey("gcp-agentless-test", accountID)+"\"}}"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "component.1.instance", "secure-scanning"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp_agentless_scanning", "component.1.service_principal_metadata", "{\"gcp\":{\"email\":\"email_value\",\"workload_identity_federation\":{\"pool_provider_id\":\"pool_provider_id_value\"}}}"),
+				),
+			},
+			{
+				ResourceName:      "sysdig_secure_cloud_auth_account.gcp_agentless_scanning",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+// XXX: add project_number to this test
```

Comment:
Pushing a backend fix for this. However, this test will require the proto fix in all envs to pass.

Reply:
Fix for bumping up proto spec merged to staging. Once latest cloudauth gets released to all regions will add back the `project_number` field in the test. Till then this PR is ready to be reviewed.

Reply:
I thought we don't need to take project_number as an input, unless you are talking about something else.

Reply:
Actually, the final decision we took was to take it as input and added the same in proto spec.

Reply:
Done and updated the test.